### PR TITLE
177374707 merge date and time

### PR DESCRIPTION
**Pull request by ToxicSam**
merge date and time using dateData 
sample output:
`{“title”:“123",“type”:“Exam”,“credit”:123,“number”:“123”,“set”:“123”,“role”:“Student”,“startDate”:“2021-1-1T00:00”,“endDate”:“2021-3-23T00:00”,“zones”:[{“questions”:[{“id”:“The_MAGICAL_MYSTERY_MACHIN”,“points”:0}]}]}`

### CODE REVIEW

**Review by timotiusvincent — COMMENTED** [review]
LGTM

**Comment by timotiusvincent** [comment]
Go ahead and merge

File: app/views/questions/index.html.erb [hunk 1]
```diff
@@ -127,14 +127,33 @@
...
           $('#export').click(function(){
             jsonData = {}
+            dateData = {} // store the date information for jsonDate
             var formData = $("#config").serializeArray();
             $.each(formData, function() {
               if (this.name == 'credit'){
                 jsonData[this.name] = new Number(this.value)
               } else {
-                jsonData[this.name] = this.value
+                switch (this.name) {
```
Comment (rchunter) [inline app/views/questions/index.html.erb:136]: This code is repeated and redundant, you can smoosh it to

case 'startDate':
case 'endDate':
case 'startTime':
case 'endTime':
    dateData[this.name] = this.value
    break;
default:
    jsonData[this.name] = this.value

as case will run until they hit a break

File: app/views/questions/index.html.erb [hunk 2]
```diff
@@ -127,14 +127,33 @@
...
+                    break;
+                  case 'startTime':
+                    dateData[this.name] = this.value
+                    break;
+                  case 'endTime':
+                    dateData[this.name] = this.value
+                    break;
+                  default:
+                    jsonData[this.name] = this.value
+                }
```
Comment (rchunter) [inline app/views/questions/index.html.erb:151]: This code is repeated a bit, you can smoosh it to

```
case 'startDate':
case 'endDate':
case 'startTime':
case 'endTime':
    dateData[this.name] = this.value
    break;
default:
    jsonData[this.name] = this.value

```
as case will run until they hit a break

**Review by chrononyan — COMMENTED** [review]
**What is the goal of this PR?**

To merge separate date and time values from the form into a single datetime before sending it to the PL backend

**Were you able to view screenshots, story cards, and or test the app?**

Yes, we were able to see the application by finding the link on the info.yml sheet. The app does load and does seem to be functioning.

**Is the PR in a shape that makes it easy to review?**

Yes, it seems to be in a bite-sized format, which makes it easy to review and understand the given goal. Function names and #div identifiers were easy to decipher. That being said, the code was a bit redundant, and could have been cleaned up just a bit. The code was also well commented. The pivotal story had no description, and even though it is a fairly straightforward piece of code, a bit more context in the description would have helped. 

**What is the status of the tests?**

There are no included tests in this pull request. Although the pull request was atomized (i.e. the feature request was small, which is great), there were no included tests to ensure that the feature was indeed working as expected.

**What is the style of the code like?**

Looking at the build configuration file for the Travis run, there doesn’t seem to be any automatic linters that were run for this PR. The Javascript code has inconsistent spacing and semicolons. That said, their code does otherwise seem to follow a standard format. The status is passing for Travis. As stated above, their code was commented and well organized (except for the placement of the Javascript) and was easily readable.

**Is there anything missing from this PR that should be included?**

As mentioned above, there was no testing included - we were also not able to confirm the change worked because the export button did not work when we tested the deployed Heroku app. This PR changes the internal representation of the datetime object, so we have no way of ensuring that this code works given that there were no explicit tests either.

**Feature Design: Are there any parts of this feature that are cause for concern? There might be nothing. Could this feature be abused by users? Does the code look safe in how it treats data? Are there any potential bugs that should be guarded against?**

It's all front end, so no data is touched and there is not a lot of room for abuse. But the string building used could construct a date that is not RFC3399 ("YYYY-MM-DDTHH:MM:SS"), which the PL docs require for dates.

---

### Overall Assessment

**What does this PR do well?**

This PR does a few things very well - it focuses on one single feature, which is great, in case something needs to be rolled back or checked. The amount of code committed is less than 30 lines, which made it easy to read and understand. Furthermore, the code was well written and mostly linted properly.

**What is something that could be improved with this PR?**

A couple of things stood out to us when we went over the PR. First, the Javascript code was added to the end of the page (and it seems like it’s still there) instead of being apportioned to its own file in the proper place. Furthermore, the functionality did not include any rspec, javascript or acceptance tests in order to ensure that the feature works as intended.

**Pull request merged**